build/release.sh

```diff
@@ -36,31 +38,25 @@ export BUILD_USER="$git_user"
 export BUILD_DATE=$( date +%Y%m%d ) # Release date is only to day-granularity
 export VERBOSE=true
 
-# Build the release binary with libpfm4 for docker container
-export GO_FLAGS="-tags=libpfm,netgo"
-build/build.sh
-
 # Build the docker image
 echo ">> building cadvisor docker image"
-gcr_tag="gcr.io/cadvisor/cadvisor:$VERSION"
-docker build -t $gcr_tag -f deploy/Dockerfile .
-
-# Build the release binary without libpfm4 to not require libpfm4 in runtime environment
-unset GO_FLAGS
-build/build.sh
-
-echo
-echo "double-check the version below:"
-echo "VERSION=$VERSION"
-echo
-echo "To push docker image to gcr:"
-echo "docker push $gcr_tag"
-echo
-echo "Release info (copy to the release page):"
-echo
-echo "Docker Image: N/A"
-echo "gcr.io Image: $gcr_tag"
-echo
-sha256sum --tag cadvisor
+image_name=${IMAGE_NAME:-"gcr.io/cadvisor/cadvisor"}
+final_image="$image_name:${VERSION}"
+
+docker buildx rm cadvisor-builder || true
+docker buildx create --name cadvisor-builder
+docker buildx use cadvisor-builder
+
+# Build binaries
+declare -A arches=( ["amd64"]="x86_64" ["arm"]="armv7l" ["arm64"]="aarch64" ["ppc64le"]="ppc64le" ["s390x"]="s390x" )
+for arch in "${!arches[@]}"; do
+  GO_ARCH="$arch" OUTPUT_NAME_WITH_ARCH="true" build/build.sh
+  arch_specific_image="${image_name}-${arch}:${VERSION}"
+  docker buildx build --platform "linux/${arch}" -f deploy/Dockerfile -t "$arch_specific_image"  --progress plain --push .
+  docker manifest create --amend "$final_image" "$arch_specific_image"
+  docker manifest annotate --os=linux --arch="$arch" "$final_image" "$arch_specific_image"
```

Comment:
just curious, for my own understanding ,why do we need to use `docker manifest annotate` here? Does docker buildx not do this automatically?

Reply:
`docker buildx build` will create a multi-arch manifest if you specify multiple platforms `--platform linux/amd64,linux/arm64` and the `--push` flag.

However, it does not tag intermediate containers like `cadvisor-arm64`. It makes it a little harder when you want to test a specific architecture locally.

Reply:
ah gotcha, thanks for the explanation!